### DISCUSSION TITLE
Deployment fail

### DISCUSSION THREAD

**Discussion by Sankito** [vercel/community#7289] [Help] [answered]
### Summary

Hi, I am running in a big trouble where i had my webapp deployed on vercel; and my computer crashed during a commit/deployment, that ruined my git repo and lost the track of my work on deployed site. 

I downloaded the last working version of my webapp and delete my repo in order to create a new one without problems (git headers and co)

I pushed my repo on my git, and tried to deploy like i did for the first webapp. I am running into this error no matter what i do, i corrected some files, everything is ok on pnpm dev, but the vercel deployment is impossible.

Nothing changed from my deployed app so I don't get why it keeps failing, while it worked fine the first time.

build:     at r (/vercel/path0/.next/server/chunks/2021.js:2:6978)
build:     at 37357 (/vercel/path0/.next/server/chunks/2021.js:1:1272)
build:     at t (/vercel/path0/.next/server/webpack-runtime.js:1:143)
build:     at 52021 (/vercel/path0/.next/server/chunks/2021.js:1:174)
build:     at t (/vercel/path0/.next/server/webpack-runtime.js:1:143)
build:     at 81422 (/vercel/path0/.next/server/app/api/auth/whoAmI/route.js:1:1451)
build:     at t (/vercel/path0/.next/server/webpack-runtime.js:1:143)
build:     at t (/vercel/path0/.next/server/app/api/auth/whoAmI/route.js:1:2177)
build:     at /vercel/path0/.next/server/app/api/auth/whoAmI/route.js:1:2230 {
build:   code: 'ERR_INVALID_URL',
build:   input: 'undefined'
build: }
build: 
build: > Build error occurred
build: Error: Failed to collect page data for /api/auth/whoAmI
build:     at /vercel/path0/node_modules/.pnpm/next@14.1.0_@babel+core@7.24.6_react-dom@18.3.1_react@18.3.1__react@18.3.1/node_modules/next/dist/build/utils.js:1258:15
build:     at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
build:   type: 'Error'
build: }
build:  ELIFECYCLE  Command failed with exit code 1.
build: ERROR: command finished with error: command (/vercel/path0/) /pnpm9/node_modules/.bin/pnpm run build exited (1)
build: command (/vercel/path0/) /pnpm9/node_modules/.bin/pnpm run build exited (1)
  Tasks:    0 successful, 1 total

### Steps to Reproduce

- simple deployment on vercel
- crash during an update
- try to push the same project in a clean new branch

**Comment by pawlean** [comment, marked as answer]
Hi @Sankito!

This sounds like an issue with your code, specifically a URL error ("ERR_INVALID_URL") Check your code to ensure that you're not passing undefined as a URL or parameter. 

Since you also mentioned that you corrected some files, double-check that you didn't introduce any breaking changes. Compare your code with the previous working version to identify any differences.

If all else fails, it's worth creating a minimal reproduction of your project, stripping away any unnecessary code, to isolate the issue. This will help you identify if the problem is specific to your code or a more general issue. I created a fresh, simple deployment on Vercel and it works fine. 

Hopefully you get to the root cause! Let me know if you have any more questions.

**Reply by Sankito** [reply]
Hey 

Ty for your answer 

I figured out that the turbo function broke on pnpm, running vercel.json without turbo works

I'll find a way to repair that

**Reply by pawlean** [reply]
Great, thanks @Sankito. Let us know if we can help with anything else. :)

[a comment by an automated account is not shown]
